test/oli_web/controllers/api/automation/automation_setup_controller_test.exs

```diff
@@ -0,0 +1,232 @@
+
+      # Make sure we cleaned up all the records
+      assert revision_count_before == Repo.one(from r in Revision, select: count(r.id))
+      assert resource_count_before == Repo.one(from r in Resource, select: count(r.id))
+    end
+
+    test "Can not tear down resources not created by automation api", %{
+      conn: conn,
+      api_key: api_key,
+      project: project
+    } do
+      author = hd(project.authors)
+
+      conn =
+        conn
+        |> put_req_header("accept", "application/json")
+        |> put_req_header("authorization", "Bearer #{api_key}")
+        |> Oli.Plugs.ValidateAPIKey.call(&Oli.Interop.validate_for_automation_setup/1)
+        |> AutomationSetupController.teardown(%{
+          "author_email" => author.email,
+          "author_password" => "unknown",
+          "educator_email" => nil,
+          "educator_password" => nil,
+          "learner_email" => nil,
+          "learner_password" => nil,
+          "project_slug" => project.slug,
+          "section_slug" => nil
+        })
+
+      response = Jason.decode!(conn.resp_body)
+
+      assert %{
+               "author_deleted" => %{"success" => false, "message" => "User not found"},
+               "educator_deleted" => %{"success" => false, "message" => "User not specififed"},
+               "learner_deleted" => %{"success" => false, "message" => "User not specififed"},
+               "project_deleted" => %{
+                 "success" => false,
+                 "message" => "Can only delete projects with no authors"
+               },
+               "section_deleted" => %{"success" => false, "message" => "No section slug provided"}
+             } = response
+    end
+  end
+
+  test "Invalid API key doesn't work", %{
+    conn: conn,
+    project: project
+  } do
+    conn =
+      conn
+      |> put_req_header("accept", "application/json")
+      |> put_req_header("authorization", "Bearer BAD-API-KEY")
+      |> Oli.Plugs.ValidateAPIKey.call(&Oli.Interop.validate_for_automation_setup/1)
+      |> AutomationSetupController.teardown(%{
+        "author_email" => nil,
+        "author_password" => nil,
+        "educator_email" => nil,
+        "educator_password" => nil,
+        "learner_email" => nil,
+        "learner_password" => nil,
+        "project_slug" => project.slug,
+        "section_slug" => nil
+      })
+
+    assert conn.status == 403
+  end
+
+  defp validate_user(email, password, user_type) do
+    config = OliWeb.Pow.PowHelpers.get_pow_config(user_type)
+
+    case Pow.Operations.get_by([{:email, email}], config) do
+      nil ->
+        {:error, "User not found"}
+
+      user ->
+        case user.__struct__.verify_password(user, password) do
+          true -> {:ok, user}
+          false -> {:error, "Credentials didn't match"}
+        end
+    end
+  end
+
+  def create_project(_) do
+    {:ok,
+     %{
+       project: Repo.insert!(Oli.Factory.project_factory())
```

Comment:
You can import the factory here and call `insert(:project)`, avoiding the call to `Repo.insert`

Reply:
Thanks! I guess it's time for me to actually go read those ExMachina docs 😄